internal/service/alarm_subscription_handler.go

```diff
@@ -358,3 +402,206 @@ func (h *alarmSubscriptionHandler) mapItem(ctx context.Context,
 	//TBD only save related attributes in the future
 	return input, nil
 }
+
+func (h *alarmSubscriptionHandler) ConfigMapAddEntry(
+	ctx context.Context, entry_key string, value string) (err error) {
+
+	//test to read the configmap
+	configmap := &corev1.ConfigMap{}
+
+	key := clnt.ObjectKey{
+		Namespace: TESTNAMESPACE,
+		Name:      TESTCONFIGMAPNAME,
+	}
+	err = h.hubClient.Get(ctx, key, configmap)
+
+	if err != nil && !apierrors.IsNotFound(err) {
+		//there is error and err is not notfound error
+		//panic("unexpected error")
+		return
+	}
+
+	saved_data := configmap.Data
+
+	//configmap does not exist
+	if saved_data == nil {
+		saved_data = make(map[string]string)
+	}
+	saved_data[entry_key] = value
+
+	configmap = &corev1.ConfigMap{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "ConfigMap",
+			APIVersion: "v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: TESTNAMESPACE,
+			Name:      TESTCONFIGMAPNAME,
+		},
+		Data: make(map[string]string),
+	}
+
+	configmap.Data = saved_data
+
+	//apply new data
+
+	//u := &unstructured.Unstructured{}
+	//u.Object, _ = runtime.DefaultUnstructuredConverter.ToUnstructured(configmap)
+	err = h.hubClient.Patch(ctx, configmap, clnt.Apply, clnt.ForceOwnership, clnt.FieldOwner("field-owner"))
+
+	return err
+}
+
+func (h *alarmSubscriptionHandler) ConfigMapReadEntry(
+	ctx context.Context, entry_key string) (value string, err error) {
+
+	//test to read the configmap
+	configmap := &corev1.ConfigMap{}
+
+	key := clnt.ObjectKey{
+		Namespace: TESTNAMESPACE,
+		Name:      TESTCONFIGMAPNAME,
+	}
+	err = h.hubClient.Get(ctx, key, configmap)
+
+	if err != nil && !apierrors.IsNotFound(err) {
+		//there is error and err is not notfound error
+		//panic("unexpected error")
+		return
+	}
+
+	if configmap.Data == nil {
+		return
+	}
+
+	value, ok := configmap.Data[entry_key]
+	if !ok {
+		err = ErrNotFound
+	}
+
+	return
+}
+
+func (h *alarmSubscriptionHandler) ConfigMapDeleteEntry(
+	ctx context.Context, entry_key string) (err error) {
+
+	//test to read the configmap
+	configmap := &corev1.ConfigMap{}
+
+	key := clnt.ObjectKey{
+		Namespace: TESTNAMESPACE,
+		Name:      TESTCONFIGMAPNAME,
+	}
+	err = h.hubClient.Get(ctx, key, configmap)
+
+	if err != nil && !apierrors.IsNotFound(err) {
+		//there is error and err is not notfound error
+		//panic("unexpected error")
+		return
+	}
+
+	if configmap.Data == nil {
+		return
+	}
+
+	_, ok := configmap.Data[entry_key]
+
+	//entry not found
+	if !ok {
+		return
+	}
+	delete(configmap.Data, entry_key)
+
+	// Note: there is a case configmap is empty shall we remove the configmap
+	err = h.hubClient.Update(ctx, configmap, &clnt.UpdateOptions{})
```

Comment:
I think we should use here the same `Patch` method that you used in `ConfigMapAddEntry`, otherwise the `Update` method may generate conflicts and we would need to detect and handle them.

Reply:
changed.